### ISSUE TITLE
App Engine Cloud Endpoints support

### ISSUE THREAD

**Issue by juriwiens** [google/google-api-nodejs-client#416]
Hi,

In the 0.x versions I was able to use this client library with my App Engine Cloud Endpoints via a custom discovery like this:

```javascript
googleapis
    .discover(myEndpointsApiName, myEndpointsApiVersion, {baseDiscoveryUrl: "https://myapp.appspot.com/_ah/api/discovery/v1/apis/"})
```

Does a solution exist to interact with Cloud Endpoints in 2.x?

**Comment by SpainTrain** [comment]
:+1: Any news on this?

**Comment by ronbeltran** [comment]
I also wanted to know how to do this in version 2.x. If not i will be using version 0.8 for custom discovery of apis.

**Comment by Gooseus** [comment]
Any sort of official response to this or a workaround to access Cloud Endpoints via node would be greatly appreciated.

**Comment by erlichmen** [comment]
I wrote the original patch to support Cloud Endpoints with this package and I'm heartbroken it didn't make it to the 1.x and 2.x releases.

**Comment by cyrus007** [comment]
Are the 0.x versions also broken, by which I mean google returns "Error 400 - Bad Request" for me ?

**Comment by lostpebble** [comment]
Is this still not working with this library? Migrating from Java endpoints to a node setup at the moment and would like to use Endpoints for communication between my node and java modules. Searching around I haven't found any solutions for using App Engine Endpoints with Nodejs as of yet...

**Comment by jmdobry** [comment]
Should be fixed by #555 

As of 5.2.0 you should be able to do:

```js
var google = require('googleapis');
google.discover('https://myapp.appspot.com/_ah/api/discovery/v1/apis', function (err, apis) {
  var myapi = google.myapi('v3');
  // use myapi
});
```

If anyone wants to try it, tell me how it goes.
